client/my-sites/earn/components/add-edit-plan-modal/index.tsx

```diff
@@ -139,6 +139,8 @@ const RecurringPaymentsPlanAddEditModal = ( {
 	const [ editedSchedule, setEditedSchedule ] = useState(
 		product?.renewal_schedule ?? PLAN_MONTHLY_FREQUENCY
 	);
+	const previousScheduleRef = useRef( product?.renewal_schedule ?? PLAN_MONTHLY_FREQUENCY );
```

Comment:
Nice use of `useRef`

Reply:
I would have written
```
const previousScheduleRef = useRef( editedSchedule );
```
as it is defined right above...

Reply:
See https://github.com/Automattic/gold/issues/248#issuecomment-1857670698, but I ultimately believe basing everything on the plan frequency is a bit "weak".
I'd rather use if the product id is set AND add a state for [the "dirtiness" of the name input](https://www.w3.org/TR/2011/WD-html5-20110525/the-input-element.html#concept-input-value-dirty-flag).

Reply:
> I would have written

Yes, that would have been better. Not sure why I didn't think of it then.

> I'd rather use if the product id is set AND add a state for the "dirtiness" of the name input.

I considered using a dirty bit as well but didn't want to track another variable for this one purpose. The refactor I had at this commit asked the question "did the title match the default title for the schedule _prior_ to the user changing the schedule?" If yes, it considered a the title unchanged, as it was identical to the default title for that schedule.

I previously thought we wanted to maintain the changing of the title for edited products as well, but after discussing this with you on our 1:1 I understand that it's perhaps better to not change the title of edited products. I was doing this with `product.title` but I can use `product.id` instead.

I'll try the `nameWasEdited` / `nameWasTouched` approach next and update the test plan.